### ISSUE TITLE
Windows newlines in diff --cached after git add with nbstripout

### ISSUE THREAD

**Issue by tgandor** [kynan/nbstripout#110]
I installed `nbstripout` without arguments (locally) in the repository.

Then I added a "dirty" notebook, say with `git add myNb.ipynb`.

Ok, doing a check with `git diff --cached` - I don't see outputs or metadata (OK!), but all the inserted lines have a "^M" (carriage return) in the diff view.

I ran `nbstripout myNb.ipynb` and git status showed no changes. Even so, I re-ran `git add myNb.ipynb` and then `git diff --cached` showed a lot of newlines.

I also have `nbdime` installed, so I saw that the `[diff ipynb]` (or sth like that) section in `.git/config` and deleted it. The ugly JSON diffs with ^M disappeared, and the structured nbdime diff shows no appended newlines (but - it may actually mask the problem).

I can't be sure what is at fault here:
- stripping while adding (adds newlines)?
- the diff configuration (or underlying command) gives spurious output?
- core.autocrlf=true doesn't work after configuring nbstripout?

I'd like to know, if this is a false alarm, or actual whitespace pollution to the repository.

**Comment by kynan** [comment]
Can you please confirm your platform, Python and nbstripout version?

**Comment by tgandor** [comment]
I'm sorry but I've been working in a setup where I wanted to commit the cell outputs together with the notebooks recently. In the next few days I'll set up nbstripout again and try to reproduce this.

**Comment by amard33p** [comment]
Windows10 == 1809
Python == 3.8.1
nbstripout == 0.3.7

```
--- a/arrays_strings/rotation/rotation_challenge.ipynb
+++ b/arrays_strings/rotation/rotation_challenge.ipynb
@@ -165,9 +165,9 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.5.0"
+   "version": "3.8.1"^M
   }
  },
  "nbformat": 4,
- "nbformat_minor": 0
+ "nbformat_minor": 1^M
 }

```

**Comment by kynan** [comment]
What is your Git config setting for `core.autocrlf` and `core.safecrlf`?

**Comment by amard33p** [comment]
> What is your Git config setting for `core.autocrlf` and `core.safecrlf`?

`$ git config core.autocrlf`
input

`$ git config core.safecrlf`
\<no output\>

**Comment by kynan** [comment]
I don't have access to a Windows system to reproduce this and I don't have a good idea what might cause this behaviour and if it's working as intended or not.

Does `nbstripout` change this behavious i.e. when you change a notebook without having `nbstripout` installed and then `git add` it, does `git diff --cached` show the `^M`?

**Comment by kynan** [comment]
@amard33p any update on this?

**Comment by amard33p** [comment]
> @amard33p any update on this?

@kynan Sorry this was quite a while back. I have since switched to colab research from Google for Jupyter notebooks. It serves my purpose.  
https://github.com/kynan/nbstripout/pull/126 looks promising though...can't verify it.

**Comment by kynan** [comment]
@ooiM has this been addressed by #126? Can we close this?

**Comment by ooiM** [comment]
@kynan Yes, I think the problem was solved.
